Source/SwiftLintBuiltInRules/Rules/Lint/UnhandledThrowingTaskRule.swift

```diff
@@ -0,0 +1,260 @@
+              } catch {
+                try bar()
+              }
+            }
+            """),
+            Example("""
+            ↓Task {
+              do {
+                try foo()
+              } catch {
+                throw BarError()
+              }
+            }
+            """)
+        ]
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
+
+private extension UnhandledThrowingTaskRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        override func visitPost(_ node: FunctionCallExprSyntax) {
+            if node.hasViolation {
+                violations.append(node.calledExpression.positionAfterSkippingLeadingTrivia)
+            }
+        }
+    }
+}
+
+private extension FunctionCallExprSyntax {
+    var hasViolation: Bool {
+        isTaskWithImplicitErrorType &&
+            doesThrow &&
+            !(isAssigned || isValueAccessed)
+    }
+
+    var isTaskWithImplicitErrorType: Bool {
+        if let typeIdentifier = calledExpression.as(IdentifierExprSyntax.self),
+           typeIdentifier.identifier.text == "Task" {
+            return true
+        }
+
+        if let specializedExpression = calledExpression.as(SpecializeExprSyntax.self),
+           let typeIdentifier = specializedExpression.expression.as(IdentifierExprSyntax.self),
+           typeIdentifier.identifier.text == "Task",
+           let lastGeneric = specializedExpression.genericArgumentClause
+            .arguments.last?.argumentType.as(SimpleTypeIdentifierSyntax.self),
+           lastGeneric.typeName == "_" {
+            return true
+        }
+
+        return false
+    }
+
+    var isAssigned: Bool {
+        guard let parent else {
+            return false
+        }
+
+        if parent.is(InitializerClauseSyntax.self) {
+            return true
+        }
+
+        if let list = parent.as(ExprListSyntax.self),
+           list.contains(where: { $0.is(AssignmentExprSyntax.self) }) {
+            return true
+        }
+
+        return false
+    }
+
+    var isValueAccessed: Bool {
+        guard let parent = parent?.as(MemberAccessExprSyntax.self) else {
+            return false
+        }
+
+        return parent.name.text == "value"
+    }
+
+    var doesThrow: Bool {
+        ThrowsVisitor(viewMode: .sourceAccurate)
+            .walk(tree: self, handler: \.doesThrow)
+    }
+}
+
+/// If the `doesThrow` property is true after visiting, then this node throws an error that is "unhandled."
+/// Try statements inside a `do` with a `catch` that handles all errors will not be marked as throwing.
+private final class ThrowsVisitor: SyntaxVisitor {
+    var doesThrow = false
+
+    override func visit(_ node: DoStmtSyntax) -> SyntaxVisitorContinueKind {
+        // If there are no catch clauses, visit children to see if there are any try expressions.
+        guard let lastCatchClause = node.catchClauses?.last else {
```

Comment:
Should we exit early and return `.skipChildren` if `doesThrow` is true?

Reply:
Good idea!